/* Copyright 2025 The TensorFlow Authors. All Rights Reserved.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
==============================================================================*/

syntax = "proto3";

package tensorflow;

import "tensorflow/core/framework/attr_value.proto";
import "tensorflow/core/framework/function.proto";

// XLA compiler options used for constructing a TfGraphToHloCompiler instance.
// Next ID: 4
message TfGraphToHloCompilerOptions {
  string device_type = 1;
  FunctionDefLibrary flib_def = 2;
  int32 graph_def_version = 3;
}

// Arguments used for a TfGraphToHloCompiler::Compile() call.
// Next ID: 4
message TfGraphToHloCompilerCompileCallArgs {
  string compile_options = 1;
  NameAttrList function = 2;
  repeated string xla_arguments = 3;
}
